src/reducers/entities/users.ts

```diff
@@ -178,7 +186,11 @@ function profiles(state: IDMappedObjects<UserProfile> = {}, action: GenericActio
             },
         };
     }
-
+    case UserTypes.PROFILE_NO_LONGER_VISIBLE: {
+        const newState = {...state};
+        delete newState[action.data.user_id];
```

Comment:
you should check that the state actually contains the userId before deleting, just to avoid returning a newState if nothing has changed.